### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by ROBERT-MCDOWELL** [Potomac/cryptimage#1]
Hi,
- do you have a command line version of your app?
- is every kind of input, framerate and size supported?
- any chance that all the encrypt decrypt process can be made in real time?
thanks

**Comment by Potomac** [comment]
Hello,

> do you have a command line version of your app?

earlier versions of cryptimage had a command line option, this mode has been removed in favor of the GUI, I am thinking about the reintroduction of this cli mode

> is every kind of input, framerate and size supported ?

cryptimage uses a ffmpeg wrapper, so all input format/framerate/codecs supported by ffmpeg will work, then cryptimage resizes automatically the input to a 4/3 ratio 768x576 pixels (black borders can be added),

there are known bugs in cryptimage when the video file has multiple audio tracks, and with video files which use the transport stream container (*.ts, *.m2ts)

> any chance that all the encrypt decrypt process can be made in real time?
> thanks

it's difficult because there are a lot of computation on pixels, and I don't use CPU extensions like MMX, SSE, java language can be slow on multimedia/video use cases,

for discret11 and syster systems the process is almost at realtime (25 fps), especially if you have a powerfull PC and use the RGB mode for colors, but the video compression step with codecs like h264 will slow-down the process, the pal/secam feature uses also a lot of the CPU, 

note that you can open the output file with VLC even if the encrypt/decrypt process is not finished

**Comment by ROBERT-MCDOWELL** [comment]
Thanks for your answer.
To be more precise I'm looking for the discret11 / syster encryption but without their strict rules.
something that encrypt the audio/video when a user upload or stream to a website, and then a browser extension that grab the video stream when played and decrypt it in real time.
My goal is to create a browser extension that grab the stream flow from the current web page like YT to encrypt and decrypt it without to alter the metas, size and quality.
on top of that I would like to integrate a web page text encryption too, but this is another subject.
if you can enjoy this kind of project so we can work together on it.

**Comment by ROBERT-MCDOWELL** [comment]
this project may help
https://github.com/joncampbell123/composite-video-simulator

**Comment by Potomac** [comment]
Ok so you want a kind of web solution, 

> To be more precise I'm looking for the discret11 / syster encryption but without their strict rules.
> something that encrypt the audio/video when a user upload or stream to a website

on server side you can use java technology (jee) for discret11/syster encryption, by using  a light version of cryptimage, as a java library (a modified version of cryptimage, with cli mode), a tomcat server must be installed :
http://tomcat.apache.org/

and with eclipse you have to create a dynamic web project in eclipse, compatible with tomcat : 
https://www.javatpoint.com/creating-jsp-in-eclipse-ide
http://blog.paumard.org/tutoriaux/eclipse-tomcat/

other solution will be to do this in PHP, with apache, if a video php library exists, for manipulating pixels

> My goal is to create a browser extension that grab the stream flow from the current web page like YT to encrypt and decrypt it without to alter the metas, size and quality.

on the client side (web navigator) I think you can use javascript for decrypting a video, this language has APIs for video and audio : 

https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Manipulating_video_using_canvas

**Comment by ROBERT-MCDOWELL** [comment]
thanks for all the suggestions. I would like to avoid any man in the middle server interaction.
so the last suggestion would be more feasible for a browser extension. I guest the trick would be to use your code and convert it in javascript right?

**Comment by ROBERT-MCDOWELL** [comment]
thanks for all the suggestions. I would like to avoid any man in the middle server interaction.
so the last suggestion would be more feasible for a browser extension. I guess the trick would be to use your code and convert it in javascript right?

**Comment by Potomac** [comment]
If you want a simplified version of discret11 and syster then the javascript port may be not very difficult,

you need to check if javascript can allow to extract a frame in the video, in order to manipulate the pixels (for doing "line shuffling", a kind of nagravision-syster), I think it's possible : 
http://www.kaizou.org/2012/09/frame-by-frame-video-effects-using-html5-and/
https://developer.mozilla.org/en-US/docs/Web/Guide/Audio_and_video_manipulation

**Comment by ROBERT-MCDOWELL** [comment]
excellent, will be helpful thanks!

**Comment by ROBERT-MCDOWELL** [comment]
which of your file should I take a look to the algorithms needed t convert?

**Comment by Potomac** [comment]
you can read /CryptImage_Dev/src/com/ib/cryptimage/core/Syster.java, SysterDec.java, SysterEnc.java, SimpleDiscret11.java,

you have also this page which explains nagravision syster algorithm : 
http://cricrac.free.fr/download/Doc/ALGORITH.HTM

check also this project : 
https://github.com/captainjack64/Nagravision-Syster

**Comment by ROBERT-MCDOWELL** [comment]
thank you very much for the links, I hope to not have too much headdick with this code :)
if I succeed to start the project so I will create a git repo, you will be welcomed  :)

**Comment by ROBERT-MCDOWELL** [comment]
forgot to ask, audio encryption is in your code too?

**Comment by Potomac** [comment]
yes, it's the file CryptImage_Dev/src/com/ib/cryptimage/core/SoundCrypt.java

**Comment by ROBERT-MCDOWELL** [comment]
Thank you Potomac!
